nixos/modules/services/hardware/auto-epp.nix

```
@@ -0,0 +1,59 @@
+{ config, lib, pkgs, ... }:
+with lib;
+
+let
+
+ cfg = config.services.auto-epp;
+ package = pkgs.auto-epp;
+
+in {
+ options = {
+
+    services.auto-epp = {
+      enable = mkEnableOption (lib.mdDoc "auto-epp for amd active pstate");
+      batState= mkOption {
+        type = types.str;
+        default = "power";
+        description = lib.mdDoc "energy_performance_preference when on battery";
+      };
+
+     acState= mkOption {
+        type = types.str;
+        default = "balance_performance";
+        description = lib.mdDoc "energy_performance_preference when plugged in";
+      };
+
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    boot.kernelParams = [
+      "amd_pstate=active"
+    ];
+
+    environment.etc."auto-epp.conf".text = ''
+     [Settings]
+     epp_state_for_AC=${cfg.acState}
+     epp_state_for_BAT=${cfg.batState}
+    '';
```

RFC 42 compliance part 2: this can now become

```suggestion
    environment.etc."auto-epp.conf".source = format.generate "auto-epp.conf" cfg.settings;
```